### PR TITLE
Make NodeMetadata a struct, move existing map[string]string to NodeMetadata.Metadata

### PR DESCRIPTION
**Pull request by tomwilkie**
Part 1 of #315

### CODE REVIEW

File: report/topology.go [hunk 1]
```diff
@@ -41,28 +41,28 @@ type EdgeMetadata struct {
...
+	Metadata
+}
 
-// Copy returns a value copy, useful for tests.
-func (nm NodeMetadata) Copy() NodeMetadata {
-	cp := make(NodeMetadata, len(nm))
-	for k, v := range nm {
-		cp[k] = v
+// Metadata is shorthand for a string:string map.
+type Metadata map[string]string
```
Comment (peterbourgon) [inline report/topology.go]: You really only want to define a new type to implement some value-add behavior via a method. I'd prefer to remove this definition and use map[string]string instead.

**Comment by peterbourgon** [comment]
Can we continue adding commits to this PR until it represents the complete feature we're driving towards, and then merge it?

File: report/topology.go (hunk 1 above)
Comment (tomwilkie) [inline report/topology.go]: I originally did that but having map[string]string everywhere was messy.  Metadata is much easier to type.

**Comment by tomwilkie** [comment]
I'd rather not; I think the change will end up being too big.  As it stands it represents a nice, self contained and self consistent change.

File: report/topology.go (hunk 1 above)
Comment (tomwilkie) [inline report/topology.go]: report.Metadata is longer than map[string]string OTOH

Comment (peterbourgon) [inline report/topology.go]: > having map[string]string everywhere was messy

It requires one less mental dereference, which is something we should (and Go does) aim to optimize for.

Comment (tomwilkie) [inline report/topology.go]: Yeah I've come round to changing them all to map[string]string now.

**Comment by peterbourgon** [comment]
As per discussion, I would rather PRs encapsulate issues (optimize for completeness) rather than checkpoints (optimize for size). But as a checkpoint, LGTM.

**Pull request merged**